File: app/java/src/main/java/com/example/production/domain/exception/PurchaseOrderNotFoundException.java
package com.example.production.domain.exception;

/**
 * 発注が見つからない例外
 */
public class PurchaseOrderNotFoundException extends DomainException {

    public PurchaseOrderNotFoundException(String orderNumber) {
        super("発注が見つかりません: " + orderNumber);
    }
}

File: app/csharp/src/ProductionManagement.Domain/Exceptions/PurchaseOrderNotFoundException.cs
namespace ProductionManagement.Domain.Exceptions;

/// <summary>
/// 発注が見つからない例外
/// </summary>
public class PurchaseOrderNotFoundException : DomainException
{
    public string OrderNumber { get; }

    public PurchaseOrderNotFoundException(string orderNumber)
        : base($"発注が見つかりません: {orderNumber}")
    {
        OrderNumber = orderNumber;
    }
}
